src/bilder/images/edxapp_v2/deploy.py

```diff
@@ -0,0 +1,368 @@
+    dest="/etc/default/tutor_permissions_tag",
+)
+
+# Preload some docker images. This will accelerate the first startup
+server.shell(
+    name=f"Preload mitodl/edxapp:{OPENEDX_RELEASE}-{EDX_INSTALLATION_NAME}",
+    commands=[
+        f"/usr/bin/docker pull mitodl/edxapp:{OPENEDX_RELEASE}-{EDX_INSTALLATION_NAME}"
+    ],
+)
+
+# Create skeleton directory structures for docker-compose
+data_directory = Path("/opt/data")
+media_directory = data_directory.joinpath("media")
+settings_directory = DOCKER_COMPOSE_DIRECTORY.joinpath("settings")
+tls_directory = DOCKER_COMPOSE_DIRECTORY.joinpath("tls")
+
+files.directory(
+    name="Create docker compose directory",
+    path=str(DOCKER_COMPOSE_DIRECTORY),
+    user="root",
+    group="root",
+    present=True,
+)
+
+files.directory(
+    name="Create data directory",
+    path=data_directory,
+    user="root",
+    group="root",
+    present=True,
+)
+files.directory(
+    name="Create media directory",
+    path=media_directory,
+    user="root",
+    group="root",
+    present=True,
+)
+files.directory(
+    name="Create settings directory",
+    path=settings_directory,
+    user="root",
+    group="root",
+    present=True,
+)
+files.directory(
+    name="Create TLS directory",
+    path=tls_directory,
+    user="root",
+    group="root",
+    present=True,
+)
+
+watched_files: list[Path] = []
+consul_templates: list[ConsulTemplateTemplate] = []
+
+# Firstly place down normal files requiring no special templating activities
+untemplated_files = {
+    "Caddyfile": settings_directory,
+}
+
+for ut_filename, dest_dir in untemplated_files.items():
+    files.put(
+        name=f"Place {ut_filename} file",
+        src=str(FILES_DIRECTORY.joinpath(ut_filename)),
+        dest=str(dest_dir.joinpath(ut_filename)),
+        mode="0664",
+    )
+    watched_files.append(dest_dir.joinpath(ut_filename))
+
+## Place down not-special consul-template files
+## Assume a .tmpl file extension that will be retained
+consul_templated_files = {
+    "docker-compose.yaml": (DOCKER_COMPOSE_DIRECTORY, "0660"),
+    ".env": (DOCKER_COMPOSE_DIRECTORY, "0660"),
+}
+
+for ct_filename, dest_tuple in consul_templated_files.items():
+    template = place_consul_template_file(
+        name=ct_filename,
+        repo_path=FILES_DIRECTORY,
+        template_path=Path(CONSUL_TEMPLATE_DIRECTORY),
+        destination_path=dest_tuple[0],
+        mode=dest_tuple[1],
+    )
+    consul_templates.append(template)
+    watched_files.append(template.destination)
+
+# Initalize the default consul configuration
+consul_configuration = {
+    Path("00-default.json"): ConsulConfig(
+        addresses=ConsulAddresses(dns="127.0.0.1", http="127.0.0.1"),
+        advertise_addr='{{ GetInterfaceIP "ens5" }}',
+        services=[],
+    )
+}
+
+# Create a few in-line consul templates for the wildcard certificate + key
+# but only do this on webnodes to limit distribution of secrets
+if node_type == WEB_NODE_TYPE:
```

Comment:
Need to add an if/else block to set `COMPOSE_PROFILE` in `/etc/default/docker-compose` to either `web` or `worker` based on `node_type` (or it could be `web,worker`, too, and run both on the same node...).